core-3/src/main/java/SpringApp.java

```diff
@@ -1,15 +1,44 @@
-// TODO copy
-// 1. Copy all created beans into this module
-// 2. Create FreeParkingLot bean and collect all available car beans inside with list. Use field wiring feature this time.
+// TODO
+// 1. Copy all created beans into this module and create two contexts
+// 1a. The first context should be loaded with scanning feature and contain 4 beans (2 cars, 2 engines)
+// 1b. The second context should be loaded with configuration class and contain 4 beans (2 cars, 2 engines).
+// REQUIRED CONDITION. Configuration context should not use direct method calls for bean wiring.
+// 2. Create FreeParkingLot bean for every context and collect all available car beans inside with list. Use field wiring feature this time, if possible.
 // 3. ParkingLot should print QUANTITY of Car beans inside.
-// 4. Create application context an print all bean names once again, using ParkingLot bean. Make sure context is created without errors
+// 4. Create application context an print all bean names once again.
+
+import config.config.ProjectConfigConfig;
+import scanning.ParkingLot.ParkingLot;
+import scanning.car.Car;
+import scanning.config.ProjectConfigScan;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import scanning.engine.Engine;
 
 public class SpringApp {
 
     public static void main(String[] args) {
         // initiate application context there
+        var context1 = new AnnotationConfigApplicationContext(ProjectConfigScan.class);
```

Comment:
remove ProjectConfig.class from here, context1 should be loaded with scanning packages feature, not with configuration class